src/flashMint/builders/wrapped.ts

```diff
@@ -0,0 +1,113 @@
+import { TransactionRequest } from '@ethersproject/abstract-provider'
+import { BigNumber } from '@ethersproject/bignumber'
+import { PopulatedTransaction } from '@ethersproject/contracts'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+import { ComponentSwapData } from '../../utils/componentSwapData'
+import { getFlashMintWrappedContract } from '../../utils/contracts'
+import { ComponentWrapData } from '../../utils/wrapData'
+import { TransactionBuilder } from './interface'
+
+export interface FlashMintWrappedBuildRequest {
+  isMinting: boolean
+  indexToken: string
+  inputOutputToken: string
+  indexTokenAmount: BigNumber
+  inputOutputTokenAmount: BigNumber
+  componentSwapData: ComponentSwapData[]
+  componentWrapData: ComponentWrapData[]
+}
+
+export class WrappedTransactionBuilder
+  implements
+    TransactionBuilder<FlashMintWrappedBuildRequest, TransactionRequest>
+{
+  constructor(private readonly provider: JsonRpcProvider) {}
+
+  async build(
+    request: FlashMintWrappedBuildRequest
+  ): Promise<TransactionRequest | null> {
+    const isValidRequest = this.isValidRequest(request)
+    if (!isValidRequest) return null
+    const {
+      componentSwapData,
+      componentWrapData,
+      indexToken,
+      indexTokenAmount,
+      inputOutputToken,
+      inputOutputTokenAmount,
+      isMinting,
+    } = request
+    const inputOutputTokenIsEth =
+      inputOutputToken === '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
```

Comment:
I don't really like this. On the frontend I usually used checking by the token's symbol e.g. `ETH`. Should we do the same here? Would required to add one more property for the build request. @ElmoArmy

Reply:
So just wanted to circle back on this and say that trying to get information on ETH is not as straight forward as checking the symbol on the token as ETH is not a token and this address is "fake" in the sense that no real contract exists at that address. The contract uses this special address to specify that the input token is ETH and it is a hardcoded value in the flashmint contract to take a different code path.  I'd prefer to see this constant be a named variable though as we should avoid magic values with the exception of constant assignment.